include/arch/arc/arch.h

```diff
@@ -206,6 +206,10 @@ extern "C" {
 
 #endif /* _ASMLANGUAGE */
 
+#if defined(__CCAC__)
+/* fixme: no _Static_assert in mwdt, make it a empty macro */
+#define _ARCH_MEM_PARTITION_ALIGN_CHECK(start, size)
+#else
```

Comment:
Could we redefine BUILD_ASSERT_MSG to be NOP instead?

Also, can we use feature-flags instead of vendor-flags. E.g. if defined(NO_STATIC_ASSERT) instead of if defined(__CCAC__).